src/utils/weighted_matching.ts

```diff
@@ -0,0 +1,87 @@
+import { assert } from "./assert";
+
+/** Weighted matching.
+ * Matches N items on the "left" side with M items on the "right" side such that every item on the "left" side is part of exactly one match, and the total cost is minimized.
+ * 
+ * Takes an array `costs[N][M]` where costs[i][j] is the cost for assigning item i on the left hand side with item j on the right hand side.
+ * 
+ * Returns the minimum total cost for a valid assignment as well as the assignment indices.
+ * In the return value, `assignments[i] = j` indicates that item `i` on the lhs was assigned to item `j` on the rhs.
+ * 
+ * Requires N <= M since otherwise not all items on the lhs can be matched.
+ *
+ * This implements the Hungarian Algorithm.
+ * Source: KACTL: https://github.com/kth-competitive-programming/kactl/blob/main/content/graph/WeightedMatching.h
+ * 
+ * See https://en.wikipedia.org/wiki/Assignment_problem
+ * See https://en.wikipedia.org/wiki/Hungarian_algorithm
+ */
+export function weighted_matching(costs: number[][]): { cost: number, assignments: Int32Array } {
```

Comment:
I can't really review this ! xD

Reply:
Hahaha, don't worry. I don't really know how it works, either. But the original C++ code is very well tested, so I think it should be fine.